ipaplatform/redhat/authconfig.py

```diff
@@ -27,8 +31,195 @@
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
 
-class RedHatAuthConfig(object):
+
+class RedHatAuthTool(object):
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RedHatAuthToolBase(object):
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        """
+        Backup the system authentication resources configuration
+        :param path: directory where the backup will be stored
+        """
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        """
+        Restore the system authentication resources configuration from a backup
+        :param path: directory where the backup is stored
+        """
+        pass
+
+    @abc.abstractmethod
+    def set_nisdomain(self, nisdomain):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthToolBase):
+
+    def _get_authselect_current_output(self):
+        try:
+            current = ipautil.run(
+                [paths.AUTHSELECT, "current"], env={"LC_ALL": "C.UTF8"})
+        except ipautil.CalledProcessError:
+            logger.debug("Current configuration not managed by authselect")
+            return None
+
+        return current.raw_output.decode()
+
+    def _parse_authselect_output(self, output_text=None):
+        """
+        Parses the output_text to extract the profile and options.
+        When no text is provided, runs the 'authselect profile' command to
+        generate the text to be parsed.
+        """
+        if output_text is None:
+            output_text = self._get_authselect_current_output()
+            if output_text is None:
+                return None
+
+        cfg_params = re.findall(
+            r"\s*Profile ID:\s*(\S+)\s*\n\s*Enabled features:\s*(.*)",
```

Comment:
I added `--raw` parameter to `authselect current` in 0.4 (already in F28). This will print non-formatted output as it was given on command line with `authselect select`. I.e.:
```
$ authselect select sssd with-smartcard
$ authselect current --raw
sssd with-smartcard
```
You can use that instead.